Detectors/Base/src/GRPGeomHelper.cxx

```diff
@@ -233,9 +235,12 @@ void GRPGeomHelper::checkUpdates(ProcessingContext& pc)
     }
     if (mRequest->askAlignments) {
       for (auto id = DetID::First; id <= DetID::Last; id++) {
+        if (!mRequest->algDetMask[id]) {
+          continue;
+        }
         std::string binding = fmt::format("align{}", DetID::getName(id));
         if (pc.inputs().getPos(binding.c_str()) < 0) {
-          return;
+          LOGP(fatal, "Alignment for detector {} was requested but no binding was found!", id);
```

Comment:
Actually, the only fix which is needed is `if (pc.inputs().getPos(binding.c_str()) < 0) continue;`: the presence of the binding and `mRequest->algDetMask[id]` is the same thing by construction, and the `finaliseCCDB` is a special virtual function which is called by the DPL framework each time the `pc.inputs().get...` provides updated CCDB object, so it can never be called if the detector was not requested (== no binding).

Reply:
much easier, thank you!